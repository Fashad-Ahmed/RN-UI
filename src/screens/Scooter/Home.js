import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Image,
  ScrollView,
} from 'react-native';
import React from 'react';
// import {useNavigation} from '@react-navigation/native';
//fonts

const Home = ({navigation}) => {
  // const navigation = useNavigation();
  return (
    <View style={styles.container}>
      <ScrollView>
        <Image source={require('./1.png')} style={styles.img} />
        <Text style={styles.title} testID={'username'}>
          Max Scooter
        </Text>
        <Text style={styles.detail}>
          Nam gravida neque quis consectetur aliquam. Suspendisse sed
          ullamcorper orci, a dictum nunc. Curabitur lobortis felis nec
          efficitur dictum. Nullam pharetra in metus et semper. Sed ornare
          tincidunt enim. Sed est quam, laoreet in bibendum a, sodales sit amet
          enim. llamcorper orci, a dictum nunc.
        </Text>
        <TouchableOpacity
          style={styles.btn}
          onPress={() => {
            console.log('presss');
            // navigation.navigate('Detail');
          }}>
          <Text style={styles.txt}>Next</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'black',
  },
  img: {
    // height: '50%',
    width: '100%',
    resizeMode: 'contain',
  },
  title: {
    color: '#FFF',
    fontSize: 45,
    marginTop: 20,
    alignSelf: 'center',
    fontFamily: 'sans-serif',
    fontWeight: 'bold',
  },
  detail: {
    alignSelf: 'center',
    color: 'red',
    marginTop: 10,
    textAlign: 'center',
    paddingHorizontal: 20,
    lineHeight: 20,
  },
  btn: {
    marginTop: 40,
    backgroundColor: 'red',
    alignSelf: 'center',
    width: '70%',
    borderRadius: 40,
  },
  txt: {
    alignSelf: 'center',
    color: '#ee',
    fontSize: 30,
    fontWeight: '400',
    color: '#FFF',
  },
});
