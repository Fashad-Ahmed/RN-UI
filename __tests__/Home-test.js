import 'react-native';
import React from 'react';
import Home from '../src/screens/Scooter/Home';
import renderer from 'react-test-renderer';

test('Home snapShot', () => {
  const snap = renderer.create(<Home />).toJSON();

  expect(snap).toMatchSnapshot();
});
