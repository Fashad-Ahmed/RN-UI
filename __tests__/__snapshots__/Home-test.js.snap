// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView>
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../src/screens/Scooter/1.png",
          }
        }
        style={
          Object {
            "resizeMode": "contain",
            "width": "100%",
          }
        }
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "sans-serif",
            "fontSize": 45,
            "fontWeight": "bold",
            "marginTop": 20,
          }
        }
      >
        Max Scooter
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "red",
            "lineHeight": 20,
            "marginTop": 10,
            "paddingHorizontal": 20,
            "textAlign": "center",
          }
        }
      >
        Nam gravida neque quis consectetur aliquam. Suspendisse sed ullamcorper orci, a dictum nunc. Curabitur lobortis felis nec efficitur dictum. Nullam pharetra in metus et semper. Sed ornare tincidunt enim. Sed est quam, laoreet in bibendum a, sodales sit amet enim. llamcorper orci, a dictum nunc.
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "red",
            "borderRadius": 40,
            "marginTop": 40,
            "opacity": 1,
            "width": "70%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#FFF",
              "fontSize": 30,
              "fontWeight": "400",
            }
          }
        >
          Next
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
